lib/screens/home/cards/cards_screen.dart

```diff
@@ -0,0 +1,112 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:shop_app/localization/index.dart';
+import 'package:shop_app/models/index.dart';
+import 'package:shop_app/router/index.dart';
+import 'package:shop_app/screens/home/cards/cards_bloc.dart';
+import 'package:shop_app/screens/home/cards/widgets/card_item.dart';
+
+@RoutePage()
+class CardsScreen extends StatelessWidget implements AutoRouteWrapper {
+  const CardsScreen({super.key});
+
+  @override
+  Widget wrappedRoute(BuildContext context) {
+    context.read<CardsBloc>().load();
+
+    return this;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    EasyLocalization.of(context);
+
+    final cardsBloc = context.read<CardsBloc>();
+
+    return Scaffold(
+      body: RefreshIndicator(
+        onRefresh: () {
+          return cardsBloc.loadAsyncFuture();
+        },
+        child: CustomScrollView(
+          slivers: [
+            SliverAppBar(
+              floating: true,
+              bottom: PreferredSize(
+                preferredSize: const Size.fromHeight(50),
+                child: Row(
+                  children: [
+                    ElevatedButton.icon(
+                      onPressed: () async {
+                        final selectedDateRange = await showDateRangePicker(
+                          context: context,
+                          // dont like it, ask
+                          firstDate: DateTime(2019, 12, 10),
```

Comment:
You can dynamicaly calculate firstDate - can be inited like this `firstDate = DateTime(now.year - 4);`